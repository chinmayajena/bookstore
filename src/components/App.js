import React from "react";
import BookStoreListView from "containers/BookStoreListView.js";
import BookStoreGridView from "containers/BookStoreGridView.js";
import AddBook from "containers/Addbook.js";
import EditBook from "containers/Editbook.js";
import { Switch, BrowserRouter as Router, Route, Link } from "react-router-dom";
import {
  Navbar,
  NavItem,
  Nav,
  Form,
  FormControl,
  Button,
  Container
} from "react-bootstrap";
import BookStoreFilter from "containers/BookStoreFilter.js";

const App = () => {
  return (
    <div className="App">
      <Router>
        <div>
          <Navbar bg="light" expand="lg">
            <Navbar.Brand href="#home">bookstore!</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
              <Link to="/" className="nav-link">
                List View
              </Link>
              <Link to="/gridview" className="nav-link">
                Grid View
              </Link>
              <Link to="/createbook" className="nav-link">
                Add Book
              </Link>
              <Form className="navbar-right ml-auto">
                <BookStoreFilter />
              </Form>
            </Navbar.Collapse>
          </Navbar>
          <Container className="mt-3 pt-5">
            <div className="container shadow p-4 mb-4 bg-white">
              <Switch>
                <Route exact path="/" component={BookStoreListView} />
                <Route path="/gridview" component={BookStoreGridView} />
                <Route path="/createbook" component={AddBook} />
                <Route path="/editbook" component={EditBook} />
                {/* when none of the above match, <NoMatch> will be rendered */}
                <Route component={BookStoreListView} />
              </Switch>
            </div>
          </Container>
          /
        </div>
      </Router>
    </div>
  );
};
export default App;
