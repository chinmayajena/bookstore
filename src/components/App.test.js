import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("<App />", () => {
  it("renders the component", () => {
    const component = shallow(<App />);
    expect(component.debug()).toMatchSnapshot();                          
    const wrapper = component.find(".App");
    expect(wrapper.length).toBe(1);
  });
});
