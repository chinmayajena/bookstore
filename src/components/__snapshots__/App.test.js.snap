// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the component 1`] = `
"<div className=\\"App\\">
  <BrowserRouter>
    <div>
      <ForwardRef(Bootstrap(undefined)) bg=\\"light\\" expand=\\"lg\\">
        <ForwardRef(Bootstrap(NavbarBrand)) href=\\"#home\\">
          bookstore!
        </ForwardRef(Bootstrap(NavbarBrand))>
        <ForwardRef(Bootstrap(NavbarToggle)) aria-controls=\\"basic-navbar-nav\\" />
        <ForwardRef(Bootstrap(NavbarCollapse)) id=\\"basic-navbar-nav\\">
          <Link to=\\"/\\" className=\\"nav-link\\" replace={false}>
            List View
          </Link>
          <Link to=\\"/gridview\\" className=\\"nav-link\\" replace={false}>
            Grid View
          </Link>
          <Link to=\\"/createbook\\" className=\\"nav-link\\" replace={false}>
            Add Book
          </Link>
          <ForwardRef(Bootstrap(Form)) inline={true} className=\\"navbar-right ml-auto\\">
            <Connect(BookStoreFilter) />
          </ForwardRef(Bootstrap(Form))>
        </ForwardRef(Bootstrap(NavbarCollapse))>
      </ForwardRef(Bootstrap(undefined))>
      <main className=\\"mt-3 pt-5\\">
        <div className=\\"container-fluids p-2 mb-1 bg-white\\">
          <Switch>
            <Route exact={true} path=\\"/\\" component={[Function: Connect]} />
            <Route path=\\"/gridview\\" component={[Function: Connect]} />
            <Route path=\\"/createbook\\" component={[Function: Connect]} />
            <Route path=\\"/editbook\\" component={[Function: Connect]} />
            <Route component={[Function: Connect]} />
          </Switch>
        </div>
      </main>
      /
    </div>
  </BrowserRouter>
</div>"
`;
